### PR TITLE
Added the ability to choose a single boat view and performed code refactoring

### PR DESCRIPTION
**Pull request by nmateusiak**
image

https://github.com/AgoraSolutions-tech/grafana-leaflet-map-visualization-panel/assets/141326063/e478cde7-6d93-4241-a0e4-a63642bf7bc2

### CODE REVIEW

**Comment by nmateusiak** [comment]
https://github.com/AgoraSolutions-tech/grafana-leaflet-map-visualization-panel/assets/141326063/4d180fd7-eb88-4092-8963-a37e57d0bdab

File: src/components/CustomControls.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  options: MapOptions;
+  objects: MovingObject[];
+  currentObjects: MovingObject[];
+  onOptionsChange:  (options: MapOptions) => void;
+}
+
+const TODAY = new Date();
+const TODAY_STRING = TODAY.toDateString();
+
+export const CustomControls = (props: Props) => {
```
Comment (mcze-thaumatec) [inline src/components/CustomControls.tsx:20]: Can we have some better name for this class?

File: src/components/MapComponent.tsx [hunk 2]
```diff
@@ -70,136 +69,101 @@ export const MapComponent: React.FC<Props> = ({ options, width, height, data, on
       ...object,
       positions: object.positions.filter((position) => isSameDay(position.timestamp, new Date(dateToDisplay))),
     }))
-    .filter((object) => object.positions.length > 0);
+    .filter((object) => object.positions.length > 0)
+    .filter(object => selectedObjectId === 'all' ? object : String(object.id) === selectedObjectId
```
Comment (mcze-thaumatec) [inline src/components/MapComponent.tsx:73]: This condition is strange. Shouldn't it be object => object.id === selectedObjectId || selectedObjectId==='all'?

**Review by mcze-thaumatec — APPROVED** [review]

File: src/components/MapContentDisplaySettings.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  options: MapOptions;
+  currentSelectedbjects: MovingObject[];
+  currentObjects: MovingObject[];
+  onOptionsChange:  (options: MapOptions) => void;
+}
+
+const TODAY = new Date();
+const TODAY_STRING = TODAY.toDateString();
+
+export const CustomControls = (props: Props) => {
```
Comment (mcze-thaumatec) [inline src/components/MapContentDisplaySettings.tsx:21]: Also name change?

**Review by mcze-thaumatec — APPROVED** [review]

**Pull request merged**